frame/election-providers/src/two_phase/mod.rs

```diff
@@ -0,0 +1,1325 @@
+	PoolSubmissionFailed,
+}
+
+impl From<crate::onchain::Error> for InternalError {
+	fn from(e: crate::onchain::Error) -> Self {
+		InternalError::OnChainFallback(e)
+	}
+}
+
+impl From<sp_npos_elections::Error> for InternalError {
+	fn from(e: sp_npos_elections::Error) -> Self {
+		InternalError::NposElections(e)
+	}
+}
+
+impl From<FeasibilityError> for InternalError {
+	fn from(e: FeasibilityError) -> Self {
+		InternalError::Feasibility(e)
+	}
+}
+
+/// Errors that can happen in the feasibility check.
+#[derive(RuntimeDebug, Eq, PartialEq)]
+pub enum FeasibilityError {
+	/// Wrong number of winners presented.
+	WrongWinnerCount,
+	/// The snapshot is not available.
+	///
+	/// This must be an internal error of the chain.
+	SnapshotUnavailable,
+	/// Internal error from the election crate.
+	NposElection(sp_npos_elections::Error),
+	/// A vote is invalid.
+	InvalidVote,
+	/// A voter is invalid.
+	InvalidVoter,
+	/// A winner is invalid.
+	InvalidWinner,
+	/// The given score was invalid.
+	InvalidScore,
+	/// An error from the data provider's feasibility check
+	DataProvider(&'static str),
+}
+
+impl From<sp_npos_elections::Error> for FeasibilityError {
+	fn from(e: sp_npos_elections::Error) -> Self {
+		FeasibilityError::NposElection(e)
+	}
+}
+
+pub use pallet::*;
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + SendTransactionTypes<Call<Self>>
+	where
+		ExtendedBalance: From<InnerOf<CompactAccuracyOf<Self>>>,
+		ExtendedBalance: From<InnerOf<OnChainAccuracyOf<Self>>>,
+	{
+		type Event: From<Event<Self>>
+			+ Into<<Self as frame_system::Config>::Event>
+			+ IsType<<Self as frame_system::Config>::Event>;
+
+		/// Currency type.
+		type Currency: ReservableCurrency<Self::AccountId> + Currency<Self::AccountId>;
+
+		/// Duration of the signed phase.
+		#[pallet::constant]
+		type SignedPhase: Get<Self::BlockNumber>;
+		/// Duration of the unsigned phase.
+		#[pallet::constant]
+		type UnsignedPhase: Get<Self::BlockNumber>;
+		/// Maximum number of singed submissions that can be queued.
+		#[pallet::constant]
+		type MaxSignedSubmissions: Get<u32>;
+
+		#[pallet::constant]
+		type SignedRewardBase: Get<BalanceOf<Self>>;
+		#[pallet::constant]
+		type SignedRewardFactor: Get<Perbill>;
+		#[pallet::constant]
+		type SignedRewardMax: Get<Option<BalanceOf<Self>>>;
+
+		#[pallet::constant]
+		type SignedDepositBase: Get<BalanceOf<Self>>;
+		#[pallet::constant]
+		type SignedDepositByte: Get<BalanceOf<Self>>;
+		#[pallet::constant]
+		type SignedDepositWeight: Get<BalanceOf<Self>>;
+
+		/// The minimum amount of improvement to the solution score that defines a solution as
+		/// "better".
+		#[pallet::constant]
+		type SolutionImprovementThreshold: Get<Perbill>;
+
+		/// The priority of the unsigned transaction submitted in the unsigned-phase
+		type UnsignedPriority: Get<TransactionPriority>;
```

Comment:
why are these not `#[pallet::constant]`?

Reply:
These are not of interest to an external tool or a UI to the best of my knowledge, because they are the configurations of the embedded OCW-miner of this pallet. I reckon any other miner will prefer using its own configs, such as `MinerMaxIterations` and such. `UnsignedPriority` Is also only relevant to the OCW-miner, so it is even more specific and.